src/dialogs/stat_dialogs/chart_options_view.pas

```diff
@@ -34,107 +32,133 @@   TChartOptionsView = class(TCustomStatDialogView)
     procedure ResetView(); override;
     procedure SetModel(DataModel: TChartOptionsModel);
     procedure ResetMinMax;
+    procedure ReplaceBoxClicked(Sender: TObject);
   end;
 
 implementation
 
+uses Graphics, math;
+
 const
   TITLE_TAG    = Ord(cbT);
   FOOTNOTE_TAG = Ord(cbF);
   XTITLE_TAG   = Ord(cbXT);
   YTITLE_TAG   = Ord(cbYT);
   COLOR_TAG    = Ord(cbC);
-
+  EXPORT_TAG   = Ord(cbE);
+  REPLACE_TAG  = Ord(cbR);
   XMIN_TAG     = Ord(cbnXMin);
   XMAX_TAG     = Ord(cbnXMax);
   YMIN_TAG     = Ord(cbnYMin);
   YMAX_TAG     = Ord(cbnYMax);
+  inputSpacing = 10;
 
 { TChartOptionsView }
 
 constructor TChartOptionsView.Create(TheOwner: TComponent);
-const
-  // not guaranteed to fit every language, but calculating it from the labels
-  // is not simple (LabelText.Width does not work)
-  labelWidth = 120;
-  dateSize   = 120;
-  inputSpacing = 10;
-
 var
-  EditText:  TCustomEdit;
-  DateText:  TDateEdit;
-  LabelText: TLabel;
+  prevControl: TWinControl;
+  LabelWidth,
+  DateWidth: Integer;
   OptLabels:    array of UTF8String = ('Main Title', 'Footnote',
-                      'X-Axis Title', 'Y-Axis Title', 'Colour Selection');
-  MinMaxLabels: array of UTF8String = ('X-Axis Minumum', 'X-Axis Maximum',
+                      'X-Axis Title', 'Y-Axis Title', 'Colour Selection',
+                      'Export to file', 'Replace graph file',
+                      'X-Axis Minumum', 'X-Axis Maximum',
                       'Y-Axis Minumum', 'Y-Axis Maximum');
   i:         Integer;
+
+function MaxWidth(AStrings: array of UTF8String): integer;
+var
+  bmp: Graphics.TBitmap;
+  i: integer;
+begin
+  result := 0;
+  for i := 0 to High(AStrings) do
+    begin
+      bmp := Graphics.TBitmap.Create;
+      result := max(bmp.Canvas.TextWidth(AStrings[i]), result);
```

Comment:
You are measuring the width of a text here but on a canvas generated right before. Since the width is dependent on font, font size, dpi, etc. I have a feeling that this can return an incorrect value if we are not careful.
Is there an actual canvas you can use instead to ensure you measure according to what the user sees on the monitor?